### PR TITLE
Add: GitHub API: Get repos info and branch protection

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Adding API calls to get Repositories of an Organization
* Adding API call to get the JSON of branch protection rules

**Why**:

* Using for automation of Repository overviews

**How**:

* Using the existing GitHub API calls from the docs

**Checklist**:

- [ ] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: pontos/github/api/organizations.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class GitHubRESTOrganizationsMixin:
+    def organisation_exists(self, orga: str) -> bool:
+        """
+        Check if a single branch in a repository exists
+
+        Args:
+            repo: GitHub repository (owner/name) to use
+            pull_request: Pull request number to check
+        """
```
Comment (bjoernricks) [inline pontos/github/api/organizations.py:27]: The docstring needs an update

File: pontos/github/api/branch.py [hunk 2]
```diff
@@ -45,3 +45,19 @@ def delete_branch(self, repo: str, branch: str):
...
         response.raise_for_status()
+
+    def branch_protection_rules(self, repo: str, branch: str):
+        """
+        Get branch protection rules for a specific repositories
+        branch
+
+        Args:
+            repo: GitHub repository (owner/name) to use
+            branch: Branch to be deleted
```
Comment (bjoernricks) [inline pontos/github/api/branch.py:56]:
```suggestion
            branch: Get protection rules for this branch
```

File: pontos/github/cmds.py [hunk 3]
```diff
@@ -153,3 +153,22 @@ def labels(terminal: Terminal, args: Namespace):
...
+    try:
+        # check if Orga is existing
+        if not git.organisation_exists(orga=args.orga):
+            terminal.error(
+                f"PR {args.orga} is not existing or authorisation failed."
+            )
+            sys.exit(1)
+        terminal.ok(f"Organization {args.orga} exists.")
+        orga_json = git.get_repositories(orga=args.orga)
+        print(orga_json)
```
Comment (bjoernricks) [inline pontos/github/cmds.py:170]: Should this print statement be kept?

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: pontos/github/cmds.py (hunk 3 above)
Comment (y0urself) [reply pontos/github/cmds.py:170]: I have used this for easier testing. I probably won't use that command at all ...

File: pontos/github/api/branch.py [hunk 4]
```diff
@@ -45,3 +45,19 @@ def delete_branch(self, repo: str, branch: str):
         api = f"/repos/{repo}/git/refs/{branch}"
         response = self._request(api, request=httpx.delete)
         response.raise_for_status()
+
+    def branch_protection_rules(self, repo: str, branch: str):
+        """
+        Get branch protection rules for a specific repositories
```
Comment (bjoernricks) [inline pontos/github/api/branch.py:51]:
```suggestion
        Get branch protection rules for a specific repository
```

File: pontos/github/api/organizations.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class GitHubRESTOrganizationsMixin:
+    def organisation_exists(self, orga: str) -> bool:
+        """
+        Check if an organization exists
+
+        Args:
+            repo: GitHub repository (owner/name) to use
+            pull_request: Pull request number to check
```
Comment (bjoernricks) [inline pontos/github/api/organizations.py:26]:
```suggestion
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]

**Review by bjoernricks — APPROVED** [review]

File: pontos/github/api/branch.py (hunk 2 above)
Comment (bjoernricks) [inline pontos/github/api/branch.py]:
```suggestion
            branch: Get protection rules for this branch
```

**Pull request merged**